File: VB/Models/ObjectsList.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Web

Public Class ObjectsList
    Private Const Key As String = "SomeValue"

    Public Function GetObjects() As List(Of MyObject)
        If HttpContext.Current.Session("Key") Is Nothing Then
            Dim list As List(Of MyObject) = New List(Of MyObject)()

            list.Add(New MyObject(1, "Test1"))
            list.Add(New MyObject(2, "Test2"))
            list.Add(New MyObject(3, "Test3"))
            list.Add(New MyObject(4, "Test4"))

            HttpContext.Current.Session("Key") = list
        End If

        Return CType(HttpContext.Current.Session("Key"), List(Of MyObject))
    End Function

    Public Sub UpdateObject(ByVal obj As MyObject)
        Dim objects As List(Of MyObject) = GetObjects()

        Dim editedObject As MyObject = objects.Where(Function(m) m.ID = obj.ID).First()

        editedObject.Name = obj.Name
    End Sub
End Class

File: CS/Models/ObjectsList.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;

namespace Example.Models {
    public class ObjectsList {
        private const String Key = "SomeValue";

        public List<MyObject> GetObjects() {
            if (HttpContext.Current.Session["Key"] == null) {
                List<MyObject> list = new List<MyObject>();

                list.Add(new MyObject(1, "Test1"));
                list.Add(new MyObject(2, "Test2"));
                list.Add(new MyObject(3, "Test3"));
                list.Add(new MyObject(4, "Test4"));

                HttpContext.Current.Session["Key"] = list;
            }

            return (List<MyObject>)HttpContext.Current.Session["Key"];
        }

        public void UpdateObject(MyObject obj) {
            List<MyObject> objects = GetObjects();
            
            MyObject editedObject = objects.Where(m => m.ID == obj.ID)
                .First();

            editedObject.Name = obj.Name;
        }
    }
}